pdf_agents/agents.py

```diff
@@ -35,11 +51,83 @@ def measurement_plan(self, point: ArrayLike) -> Tuple[str, List, Dict]:
         plan_kwargs : dict
             Dictionary of keyword arguments to pass the plan, from a point to measure.
         """
-        return "agent_sample_count", [self._motor, point, self._exposure], dict(sample_number=0)
+        # TODO: get sample number from redis
+        return "agent_sample_count", [self.motor_name, point, self.exposure_time], dict(sample_number=0)
 
     def unpack_run(self, run) -> Tuple[Union[float, ArrayLike], Union[float, ArrayLike]]:
-        y = np.array(run.primary.data["chi_I"][0])
-        return run.start[self._motor][self._motor]["value"], y
+        y = run.primary.data[self.data_key].read().flatten()
+        if self.roi is not None:
+            q = np.array(run.primary.data[self.data_key]).flatten()
+            idx_min = np.where(q < self.roi[0])[0][-1] if len(np.where(q < self.roi[0])[0]) else None
+            idx_max = np.where(q > self.roi[1])[0][-1] if len(np.where(q > self.roi[1])[0]) else None
+            y = y[idx_min:idx_max]
+        return run.start[self.motor_name][self.motor_name]["value"], y
```

Comment:
@DanOlds I'll need particular attention from you here.

Reply:
So this looks like it will work.  Minor suggestion - you might want to change the variable `q` to `x` just to avoid confusion. As I understand it, we can point any datastream at this - not just I(q).